### DISCUSSION TITLE
MQTT Statistics not being shown

### DISCUSSION THREAD

**Discussion by jlanzarotta** [rabbitmq/rabbitmq-server#12381] [Questions] [answered]
### Community Support Policy

- [X] I have read [RabbitMQ's Community Support Policy](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md)
- [X] I agree to provide all relevant information (versions, logs, rabbitmq-diagnostics output, detailed reproduction steps)

### RabbitMQ version used

4.0.0-rc.1

### Erlang version used

26.2.x

### Operating system (distribution) used

Windows 10

### How is RabbitMQ deployed?

Windows installer

### rabbitmq-diagnostics status output

N/A

### Logs from node 1 (with sensitive values edited out)

N/A

### Logs from node 2 (if applicable, with sensitive values edited out)

See https://www.rabbitmq.com/docs/logging to learn how to collect logs

### Logs from node 3 (if applicable, with sensitive values edited out)

See https://www.rabbitmq.com/docs/logging to learn how to collect logs

```
### rabbitmq.conf

listeners.tcp.default = 5672
```

### Steps to deploy RabbitMQ cluster

N/A

### Steps to reproduce the behavior in question

1. Enable the management console plugin. `rabbitmq-plugins enable rabbitmq_management`
2. Enable the MQTT plugin. `rabbitmq-plugins enable rabbitmq_mqtt`
3. Start subscribing to MQTT topics.
4. Start publishing MQTT messages on those topics.

Messages are being sent and received, but no statistics are being shown on the Overview page.

![image](https://github.com/user-attachments/assets/eca1129f-3d51-47af-a518-33e98863ad15)

Also, when looking at the Queue, no information is available...

![image](https://github.com/user-attachments/assets/5677c515-3d60-4599-94cd-81fc05d50c33)

I believe this to be odd since I was uses to seeing information on these pages when using AMQP...

Thoughts?

### advanced.config

See https://www.rabbitmq.com/docs/configure#config-location to find advanced.config file location

### Application code

```python
# relevant messaging-related parts of your code go here
```

### Kubernetes deployment file

```yaml
# Relevant parts of K8S deployment that demonstrate how RabbitMQ is deployed
```

**Comment by michaelklishin** [comment]
There is a [dedicated documentation section](https://www.rabbitmq.com/docs/mqtt#metrics) in the MQTT guide. Some metrics are AMQP 0-9-1 specific and with [native MQTT](https://www.rabbitmq.com/blog/2023/03/21/native-mqtt) starting with 3.12, MQTT connections no longer rely on AMQP 0-9-1 and thus those metrics are never emitted.

Use [Prometheus and Grafana](https://www.rabbitmq.com/docs/prometheus). They have been the recommended option for monitoring for years.

**Reply by jlanzarotta** [reply, marked as answer]
Thanks for the information.  That was what I needed.  I understand using Prometheus and Grafana for monitor, but it was always nice seeing the charts directly in RabbitMQ.
